sormas-ui/src/main/java/de/symeda/sormas/ui/email/ExternalBulkEmailOptionsForm.java

```diff
@@ -0,0 +1,199 @@
+			getContent().addComponent(buildUploadButton(), UPLOAD_LOC);
+		}
+
+		if (documentRelatedEntityType != null) {
+			attachedDocumentsField = addField(ExternalEmailOptionsWithAttachmentsDto.ATTACHED_DOCUMENTS, MultiSelectFiles.class);
+			attachedDocumentsField.setDeleteAttachmentCallBack(SormasUI::refreshView);
+		}
+	}
+
+	private Button buildUploadButton() {
+		VerticalLayout uploadLayout = new VerticalLayout();
+		uploadLayout.setSpacing(true);
+		uploadLayout.setMargin(true);
+		uploadLayout.addStyleName(CssStyles.LAYOUT_MINIMAL);
+
+		mainButton =
+			ButtonHelper.createIconPopupButton(Captions.documentUploadDocument, VaadinIcons.PLUS_CIRCLE, uploadLayout, ValoTheme.BUTTON_PRIMARY);
+
+		boolean multipleUpload = UiUtil.enabled(FeatureType.DOCUMENTS_MULTI_UPLOAD);
+		AtomicInteger noOfUploadFiles = new AtomicInteger();
+		final boolean[] limitFiveInfoPopUpShown = {
+			false };
+
+		UploadStateWindow uploadStateWindow = new UploadStateWindow();
+		MultiFileUpload multiFileUpload = new DocumentMultiFileUpload(() -> {
+			mainButton.setButtonClickTogglesPopupVisibility(false);
+			mainButton.setClosePopupOnOutsideClick(false);
+		}, (inputStream, fileName, mimeType, length, filesLeftInQueue) -> {
+			int attachedFilesBeforeNextAdd = attachedDocumentsField.getSelectedItemsWithCaption().size();
+			noOfUploadFiles.set(attachedFilesBeforeNextAdd);
+			noOfUploadFiles.getAndIncrement();
+
+			List<String> acceptedFileExtensions = Arrays.asList(".docx", ".pdf", ".jpg", ".png", ".gif");
+
+			String fileExtension = FacadeProvider.getDocumentFacade().getFileExtension(fileName);
+
+			if (acceptedFileExtensions.contains(fileExtension)) {
+				if (noOfUploadFiles.get() <= MAX_ATTACHMENT_NUMBER) {
+					DocumentDto document = DocumentDto.build();
+					document.setName(fileName);
+					document.setMimeType(mimeType);
+					document.setSize(length);
+					document.setUploadingUser(FacadeProvider.getUserFacade().getCurrentUser().toReference());
+					DocumentRelatedEntitiesDto documentRelatedEntitiesDto = new DocumentRelatedEntitiesDto();
+					documentRelatedEntitiesDto.setRelatedEntityType(documentRelatedEntityType);
+
+					attachedDocumentsField.addSelectedItemWithCaption(new AttachementReferenceDto(document, getContent(inputStream)), fileName);
+				} else {
+					if (noOfUploadFiles.get() == 6 && !limitFiveInfoPopUpShown[0]) {
+						limitFiveInfoPopUpShown[0] = true;
+						VaadinUiUtil.showSimplePopupWindow(
+							I18nProperties.getString(Strings.headingBulkEmailMaxAttachedFiles),
+							I18nProperties.getString(Strings.messageBulkEmailTooManySelectedAtachments),
+							ContentMode.HTML,
+							620);
+					}
+				}
+			} else {
+				StringBuilder fileType = new StringBuilder();
+				acceptedFileExtensions.forEach(attachedFileAcceptedExtension -> {
+					fileType.append(attachedFileAcceptedExtension);
+					fileType.append(", ");
+				});
+				fileType.replace(fileType.length() - 2, fileType.length() - 1, "");
```

Comment:
what does this replace do?

Reply:
it removes the last coma from the above forEach